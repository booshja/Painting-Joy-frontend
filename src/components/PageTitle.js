// dependencies
import React from "react";
import styled from "styled-components";

const StyledTitle = styled.h1`
    text-align: center;
    font-family: "Barlow Condensed", sans-serif;
    font-size: 2.4rem;
    font-weight: 700;
    letter-spacing: 1px;
    text-transform: uppercase;
    padding: 1rem 0;
`;

const PageTitle = ({ title }) => {
    return <StyledTitle>{title}</StyledTitle>;
};

export default PageTitle;
