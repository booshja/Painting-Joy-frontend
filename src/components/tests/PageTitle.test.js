import { render } from "@testing-library/react";
import PageTitle from "../PageTitle";

const child = <h1>Test!</h1>;

/** Smoke Test */
it("renders without crashing", () => {
    render(<PageTitle>{child}</PageTitle>);
});

/** Snapshot Test */
it("matches snapshot", () => {
    const { asFragment } = render(<PageTitle>{child}</PageTitle>);
    expect(asFragment()).toMatchSnapshot();
});

/** RTL Tests */
it("renders correctly", () => {
    const { getByText } = render(<PageTitle>{child}</PageTitle>);

    getByText("Test!");
});
