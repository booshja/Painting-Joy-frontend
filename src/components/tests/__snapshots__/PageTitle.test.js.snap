// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <h1
    class="sc-bdnxRM hbRqTx"
  />
  <h1>
    Test!
  </h1>
</DocumentFragment>
`;
